Add tool for codegen removal (#422)
ostis-ai/sc-machine: docs/sc-tools/codegen_removal.md

# codegen-removal

> This is a program that translates sc-machine usage from old format into new format.

Reviewer: ```suggestion
This is a program that translates all agents implemented on the old API supported on sc-machine versions <= 0.9.0 to the new API.
```

Author: not only agents are translated, I added list with everything that is translated

> to run you can use script

Reviewer: ```suggestion
To run you can use this script
```

```shell

> ./scripts/codegen-removal/remove_codegen.sh /absolute/path/to/folder/with/project

Reviewer: It can be relative path?

Author: no it cannot be a relative path

Reviewer: why?

Author: script may be invoked from anywhere and relative path for that place may differ from relative path for script location and it is not necessary to support that

Reviewer: I think support for the relative path should be made. We support it everywhere

Author: ![image](https://github.com/user-attachments/assets/91dd5dac-30ab-42bf-8147-3a9ab8cad842)
not everywhere

Author: turns out it did work with not absolute paths

```

> there may be a situation when this program won't be able to replace code correctly, these places are marked with `//todo(codegen-removal):`

Reviewer: ```suggestion
There may be a situation when this program won't be able to replace code correctly, these code places are marked with `//todo(codegen-removal):`
```
